model/godleyIcon.h

```diff
@@ -33,16 +34,43 @@
 namespace minsky
 {
   class GodleyTableEditor;
-
-  class GodleyIcon: public ItemT<GodleyIcon>
+  
+  // Godley variables broken out in a separate structure, as copying is non-default
+  struct GodleyVars
   {
-    /// for placement of bank icon within complex
-    float flowMargin=0, stockMargin=0, iconWidth=100, iconHeight=100;
-    /// icon scale is adjusted when Godley icon is resized
-    float m_iconScale=1;
-    CLASSDESC_ACCESS(GodleyIcon);
-    friend class SchemaHelper;
+  public:	    
+    typedef std::vector<VariablePtr> Variables;
+    const Variables& flowVars() const {return m_flowVars;}
+    const Variables& stockVars() const {return m_stockVars;}    	  
+	  
+    GodleyVars() {}
+    virtual ~GodleyVars() {}
+    // copy operations not deleted to allow ItemT<GodleyIcon> to compile
+    GodleyVars(const GodleyVars& x) {const_cast<GodleyVars&>(x).update();}
+    GodleyVars& operator=(const GodleyVars&) {update(); return *this;}
+    //classdesc::Exclude<std::weak_ptr<GodleyIcon>> self; ///< weak ref to this    
+    classdesc::Exclude<std::shared_ptr<GodleyIcon>> self;
+    
+    void clear() {
+      m_flowVars.clear();
+      m_stockVars.clear();
+    }    
+
+    /// updates the variable lists with the Godley table
+    virtual void update();            
```

Comment:
Make this method pure virtual (ie =0), and provide the implementation as GodleyIcon::update